ledger/sandbox-classic/src/test/suite/scala/platform/sandbox/stores/ledger/sql/SqlLedgerSpec.scala

```diff
@@ -273,65 +273,217 @@ final class SqlLedgerSpec
-      ledgerId: LedgerId,
-      validatePartyAllocation: Boolean,
-  ): Future[Ledger] =
-    createSqlLedger(ledgerId, List.empty, validatePartyAllocation)
+  private val retryStrategy = RetryStrategy.exponentialBackoff(10, 12.millis)
 
-  private def createSqlLedger(
-      packages: List[DamlLf.Archive],
-      validatePartyAllocation: Boolean,
-  ): Future[Ledger] =
-    createSqlLedger(None, None, packages, validatePartyAllocation)
+  private def eventually[T](f: => Future[T]): Future[T] = retryStrategy.apply((_, _) => f)
 
-  private def createSqlLedger(
-      ledgerId: String,
-      packages: List[DamlLf.Archive],
-      validatePartyAllocation: Boolean,
-  ): Future[Ledger] = {
-    val assertedLedgerId: LedgerId = LedgerId(Ref.LedgerString.assertFromString(ledgerId))
-    createSqlLedger(assertedLedgerId, packages, validatePartyAllocation)
-  }
+  private def createSqlLedger(): Future[Ledger] =
+    createSqlLedger(ledgerId = None, participantId = None, packages = List.empty)
 
-  private def createSqlLedger(
-      ledgerId: LedgerId,
-      packages: List[DamlLf.Archive],
-      validatePartyAllocation: Boolean,
-  ): Future[Ledger] =
-    createSqlLedger(Some(ledgerId), None, packages, validatePartyAllocation)
+  private def createSqlLedger(ledgerId: String): Future[Ledger] =
+    createSqlLedger(ledgerId = ledgerId, packages = List.empty)
 
-  private def createSqlLedgerWithParticipantId(
-      participantId: ParticipantId,
-      validatePartyAllocation: Boolean,
-  ): Future[Ledger] =
-    createSqlLedger(None, Some(participantId), List.empty, validatePartyAllocation)
+  private def createSqlLedger(ledgerId: LedgerId): Future[Ledger] =
+    createSqlLedger(ledgerId = Some(ledgerId), participantId = None, packages = List.empty)
 
-  private def makeParticipantId(id: String): ParticipantId =
-    ParticipantId(Ref.ParticipantId.assertFromString(id))
+  private def createSqlLedger(packages: List[DamlLf.Archive]): Future[Ledger] =
+    createSqlLedger(ledgerId = None, participantId = None, packages = packages)
 
-  private val DefaultParticipantId = makeParticipantId("test-participant-id")
+  private def createSqlLedger(ledgerId: String, packages: List[DamlLf.Archive]): Future[Ledger] = {
+    val ledgerIdString: String = Ref.LedgerString.assertFromString(ledgerId)
+    val assertedLedgerId: LedgerId = LedgerId(ledgerIdString)
+    createSqlLedger(ledgerId = Some(assertedLedgerId), participantId = None, packages = packages)
+  }
+
+  private def createSqlLedgerWithParticipantId(participantId: ParticipantId): Future[Ledger] =
+    createSqlLedger(ledgerId = None, participantId = Some(participantId), packages = List.empty)
 
   private def createSqlLedger(
       ledgerId: Option[LedgerId],
       participantId: Option[ParticipantId],
       packages: List[DamlLf.Archive],
-      validatePartyAllocation: Boolean,
+      timeProvider: TimeProvider = TimeProvider.UTC,
+      enableAppendOnlySchema: Boolean = true,
```

Comment:
:+1: This is a good default, as we intend to phase out the mutable schema anyway, so eventually all test cases should be using `enableAppendOnlySchema = true`.